### ISSUE TITLE
[Suggestion] Cooperative request

### ISSUE THREAD

**Issue by phoriah** [Saalvage/BetterDiscordAddons#2]
I have found another group of people trying to fix this big issue.
[here](https://github.com/thunderstorm010/ShowHiddenChannels/)
All I'm saying is if you could potentially team up with them and unite your efforts I think that'd be great.
If more similar projects arise I will try to notify you and them.
Just a suggestion!
Thanks, P.

**Comment by Saalvage** [comment]
Thank you for bringing this to my intention! That'd definitely be cool. Paging @xAstroBoy

**Comment by mutageneral** [comment]
Saalvage's version of the add-on is much better, as it doesn't require its own version of 0BDFDB, which [breaks all add-ons using it](https://github.com/xAstroBoy/ShowHiddenChannels/issues/8#issuecomment-1237351847). The maintainer of the other fork seems uncooperative to actually fix the problem and just asks us to "switch to replugged" (an alternative to better discord) as a crutch.

Thank you for this fork @Saalvage ❤
And thank you @phoriah to made me find this fork by opening a similar issue in the other repo.

**Comment by mutageneral** [comment]
The other repo seems to have been deleted. This was not the intention of my comments.

@teisseire117 If you could not see my comment before the repo was deleted, I said I found a fix to the issues we both had by using this fork of instead.

**Comment by xAstroBoy** [comment]
Yeah, i deleted because this fork already has a better fix than my repo, but as i mentioned, i moved to replugged and installed this

https://github.com/discord-modifications/show-hidden-channels

**Comment by mutageneral** [comment]
I found another repo with the same goal to continue ShowHiddenChannels:
https://github.com/Minixsas/ShowHiddenChannelsFix
Could be cool to cooperate with them? Ping to the maintainers: @dimaguy & @Minixsas

****

Also, @Saalvage, they mention this on their readme:
> In it's current state it won't survive main library updates

Is this true for this add-on too? Next 0BDFDB update we are screwed?

**Comment by xAstroBoy** [comment]
> I found another repo with the same goal to continue ShowHiddenChannels: https://github.com/Minixsas/ShowHiddenChannelsFix Could be cool to cooperate with them? Ping to the maintainers: @dimaguy & @Minixsas
> 
> Also, @Saalvage, they mention this on their readme:
> 
> > In it's current state it won't survive main library updates
> 
> Is this true for this add-on too? Next 0BDFDB update we are screwed?

Because discord updates tend to break the library, hence i always suggest to move to replugged and install the replugged one's and for BD plugins, use bdcompat, as it allows them to work, just dont use devilbro's ones.

**Comment by dimaguy** [comment]
I suggest yall take a chill pill, hardcoding stuff from the data file will solve it for now, but upcoming updates can easily screw up the strat, my suggestion is a rewrite but I aim't got the time nor skill for that

**Comment by xAstroBoy** [comment]
> I suggest yall take a chill pill, hardcoding stuff from the data file will solve it for now, but upcoming updates can easily screw up the strat, my suggestion is a rewrite but I ain't got the time nor skill for that

Yeah, a rewrite to break free of the 0BDB library is a better way for showhiddenchannel to survive all the discord updates.

**Comment by dimaguy** [comment]
It wasn't uncalled for, there were reasons behind it. But after all, its free stuff. We are not entitled to it by any means

**Comment by teisseire117** [comment]
> The other repo seems to have been deleted. This was not the intention of my comments.
> 
> @teisseire117 If you could not see my comment before the repo was deleted, I said I found a fix to the issues we both had by using this fork of instead.

Thank you but this doesn't work, it has the same issue as the one from the previous repository, but they had fixed it somehow.

Basically it still auto-deletes itself

**Comment by teisseire117** [comment]
> > I found another repo with the same goal to continue ShowHiddenChannels: https://github.com/Minixsas/ShowHiddenChannelsFix Could be cool to cooperate with them? Ping to the maintainers: @dimaguy & @Minixsas
> > Also, @Saalvage, they mention this on their readme:
> > > In it's current state it won't survive main library updates
> > 
> > 
> > Is this true for this add-on too? Next 0BDFDB update we are screwed?
> 
> Because discord updates tend to break the library, hence i always suggest to move to replugged and install the replugged one's and for BD plugins, use bdcompat, as it allows them to work, just dont use devilbro's ones.

The thing is Devilbro made over 50 plug-ins and people use a lot of them, they definitely make the Discord experience a lot more enjoyable

**Comment by xAstroBoy** [comment]
> > > I found another repo with the same goal to continue ShowHiddenChannels: https://github.com/Minixsas/ShowHiddenChannelsFix Could be cool to cooperate with them? Ping to the maintainers: @dimaguy & @Minixsas
> > > Also, @Saalvage, they mention this on their readme:
> > > > In it's current state it won't survive main library updates
> > > 
> > > 
> > > Is this true for this add-on too? Next 0BDFDB update we are screwed?
> > 
> > 
> > Because discord updates tend to break the library, hence i always suggest to move to replugged and install the replugged one's and for BD plugins, use bdcompat, as it allows them to work, just dont use devilbro's ones.
> 
> The thing is Devilbro made over 50 plug-ins and people use a lot of them, they definitely make the Discord experience a lot more enjoyable

most of them cause instability, there's some plugins being rewritten for replugged/powercord.

**Comment by Saalvage** [comment]
The way I see it: I assume 0BDB abstracts away a large portion of the underlying stuff within Discord that might break between updates, meaning sticking with 0BDB will actually allow us to survive longer. Especially when considering that with each update changing 0BDB's API we have a huge amount of other plugins to observe the plugin conversion process (all of DevilBro's plugins), meaning converting ourselves might not be so hard!

"Rewriting" SHC would probably just come down to inlining most 0BDB's methods it's using and I personally do not believe that to be the best way forward.

**Comment by xAstroBoy** [comment]
> The way I see it: I assume 0BDB abstracts away a large portion of the underlying stuff within Discord that might break between updates, meaning sticking with 0BDB will actually allow us to survive longer. Especially when considering that with each update changing 0BDB's API we have a huge amount of other plugins to observe the plugin conversion process (all of DevilBro's plugins), meaning converting ourselves might not be so hard!
> 
> "Rewriting" SHC would probably just come down to inlining most 0BDB's methods it's using and I personally do not believe that to be the best way forward.

Take a look at https://github.com/discord-modifications/show-hidden-channels , it doesn't rely on 0BDB library

**Comment by Saalvage** [comment]
On a side note: I'd like to keep the discussion surrounding all of the drama to remain off this repo, as it's not really the place for it. feel free to head over to [/r/OpenBetterDiscord](https://www.reddit.com/r/OpenBetterDiscord/) to discuss that and keep this repo to code related matters, thanks!

**Comment by mutageneral** [comment]
> > The other repo seems to have been deleted. This was not the intention of my comments.
> > @teisseire117 If you could not see my comment before the repo was deleted, I said I found a fix to the issues we both had by using this fork of instead.
> 
> Thank you but this doesn't work, it has the same issue as the one from the previous repository, but they had fixed it somehow.
> 
> Basically it still auto-deletes itself

What I did to fix the issue is to delete 0BDFDB & ShowHiddenChannels from my plugin folder. Then I took https://github.com/Saalvage/BetterDiscordAddons/blob/master/Plugins/ShowHiddenChannelsV2/ShowHiddenChannelsV2.plugin.js and put it in my plugin folder and enabled it and reloaded discord. On reload, it prompted me to install the latest version of 0BDFDB. I did and restarted and now everything works flawlessly.

**Comment by Saalvage** [comment]
> > The way I see it: I assume 0BDB abstracts away a large portion of the underlying stuff within Discord that might break between updates, meaning sticking with 0BDB will actually allow us to survive longer. Especially when considering that with each update changing 0BDB's API we have a huge amount of other plugins to observe the plugin conversion process (all of DevilBro's plugins), meaning converting ourselves might not be so hard!
> > "Rewriting" SHC would probably just come down to inlining most 0BDB's methods it's using and I personally do not believe that to be the best way forward.
> 
> Take a look at https://github.com/discord-modifications/show-hidden-channels , it doesn't rely on 0BDB library

But I believe the underlying API for Powercord is different from BetterDiscord?

Actually, does Powercord have good docs? Because the ones for BetterDiscord are absolutely horrendous.

**Comment by xAstroBoy** [comment]
> > > The way I see it: I assume 0BDB abstracts away a large portion of the underlying stuff within Discord that might break between updates, meaning sticking with 0BDB will actually allow us to survive longer. Especially when considering that with each update changing 0BDB's API we have a huge amount of other plugins to observe the plugin conversion process (all of DevilBro's plugins), meaning converting ourselves might not be so hard!
> > > "Rewriting" SHC would probably just come down to inlining most 0BDB's methods it's using and I personally do not believe that to be the best way forward.
> > 
> > 
> > Take a look at https://github.com/discord-modifications/show-hidden-channels , it doesn't rely on 0BDB library
> 
> But I believe the underlying API for Powercord is different from BetterDiscord?
> 
> Actually, does Powercord have good docs? Because the ones for BetterDiscord are absolutely horrendous.
> 
> If they do I might just switch myself!

feel free to ask in their discord https://replugged.dev/installation

**Comment by teisseire117** [comment]
> > > The other repo seems to have been deleted. This was not the intention of my comments.
> > > @teisseire117 If you could not see my comment before the repo was deleted, I said I found a fix to the issues we both had by using this fork of instead.
> > 
> > 
> > Thank you but this doesn't work, it has the same issue as the one from the previous repository, but they had fixed it somehow.
> > Basically it still auto-deletes itself
> 
> What I did to fix the issue is to delete 0BDFDB & ShowHiddenChannels from my plugin folder. Then I took https://github.com/Saalvage/BetterDiscordAddons/blob/master/Plugins/ShowHiddenChannelsV2/ShowHiddenChannelsV2.plugin.js and put it in my plugin folder and enabled it and reloaded discord. On reload, it prompted me to install the latest version of 0BDFDB. I did and restarted and now everything works flawlessly.

That works thank you.

**Comment by xAstroBoy** [comment]
> MessageLoggerV2 seems to be good example of how plugin survives after being exiled

yeah and they even say that it will die LMAO

**Comment by mutageneral** [comment]
@xAstroBoy what is the difference between powercord and replugged? replugged seems to be a fork of powercord.

**Comment by xAstroBoy** [comment]
> @xAstroBoy what is the difference between powercord and replugged? replugged seems to be a fork of powercord.

is a fork of powercord, is the same base, because powercord is obsolete.

**Comment by Minixsas** [comment]
> > The way I see it: I assume 0BDB abstracts away a large portion of the underlying stuff within Discord that might break between updates, meaning sticking with 0BDB will actually allow us to survive longer. Especially when considering that with each update changing 0BDB's API we have a huge amount of other plugins to observe the plugin conversion process (all of DevilBro's plugins), meaning converting ourselves might not be so hard!
> > "Rewriting" SHC would probably just come down to inlining most 0BDB's methods it's using and I personally do not believe that to be the best way forward.
> 
> Take a look at https://github.com/discord-modifications/show-hidden-channels , it doesn't rely on 0BDB library

yes but how do you install that

**Comment by xAstroBoy** [comment]
> Quote reply

use replugged and use git clone in the replugged plugins folder.

**Comment by Minixsas** [comment]
> I found another repo with the same goal to continue ShowHiddenChannels: https://github.com/Minixsas/ShowHiddenChannelsFix Could be cool to cooperate with them? Ping to the maintainers: @dimaguy & @Minixsas
> 
> Also, @Saalvage, they mention this on their readme:
> 
> > In it's current state it won't survive main library updates
> 
> Is this true for this add-on too? Next 0BDFDB update we are screwed?

when he updates the library we can most likely apply the same patch to the new library as we did to the old one

**Comment by Minixsas** [comment]
I already got banned on bd discord for sending people the shc fix

**Comment by mutageneral** [comment]
you should have DM them

**Comment by xAstroBoy** [comment]
> you should have DM them

some people snitch to the BD staff lol

**Comment by Saalvage** [comment]
Imagine censoring a plugin, only thing more sad is snitching that someone is circumventing the censorship of said plugin..

**Comment by Minixsas** [comment]
I don't understand why cant he just post updates for the plugin but unofficially on some different github

**Comment by Minixsas** [comment]
I understand why they dont want to have anything to do with the plugin, but banning people for mentioning it? that's big line they crossed. Bd is already against discords tos and so why do they even fucking care

**Comment by mutageneral** [comment]
> > you should have DM them
> 
> some people snitch to the BD staff lol

So there are spies that are acting like someone asking for a SHC fork (which could also be bannable) and then they report it to admins? Are the spies the moderators on their alt accounts? Because why would anyone in their right mind be doing this otherwise... If you are actually looking for a SHC fork, you will be grateful... If you are not, then why would you gamble with getting banned by faking asking for SHC forks...

**Comment by Minixsas** [comment]
Same with message logger, there are dozens of bots that log messages but I cant log messages myself? i mean wtf

**Comment by teisseire117** [comment]
> I don't understand why cant he just post updates for the plugin but unofficially on some different github

He didn't remove the plug-in from people's folder because of new BetterDiscord TOS. BD's admins themselves said they didn't require him to do so.

One of their staff told me he was done being mentioned about this specific plug-in, so it was most likely the reason he removed it from everyone.

I think anyway he didn't acted in a very mature way, but that's whatever let's just move on from this guy

**Comment by Saalvage** [comment]
> 

I think it's not as bad as you think it to be, they probably just dmed people unsolicitedly and they reported that, still super sad.

**Comment by Minixsas** [comment]
> > I don't understand why cant he just post updates for the plugin but unofficially on some different github
> 
> He didn't remove the plug-in from people's folder because of new BetterDiscord TOS. BD's admins themselves said they didn't require him to do so.
> 
> One of their staff told me he was done being mentioned about this specific plug-in, so it was most likely the reason he removed it from everyone.
> 
> I think anyway he didn't acted in a very mature way, but that's whatever let's just move on from this guy

no they did remove it if you have that plugin in version 3.1.1 and start discord it will wipe the whole file of the plugin and it will make it to write the message they literally force wiped the plugin from your pc

**Comment by teisseire117** [comment]
> > > I don't understand why cant he just post updates for the plugin but unofficially on some different github
> > 
> > 
> > He didn't remove the plug-in from people's folder because of new BetterDiscord TOS. BD's admins themselves said they didn't require him to do so.
> > One of their staff told me he was done being mentioned about this specific plug-in, so it was most likely the reason he removed it from everyone.
> > I think anyway he didn't acted in a very mature way, but that's whatever let's just move on from this guy
> 
> no they did remove it if you have that plugin in version 3.1.1 and start discord it will wipe the whole file of the plugin and it will make it to write the message they literally force wiped the plugin from your pc

That's the guy who made the plug-in who did it, not BD's admins

**Comment by Saalvage** [comment]
Man, I can understand being annoyed at bugs, but acting unkind is just a no-go imo, it's not the fault of your users that your software is bugged!

**Comment by teisseire117** [comment]
> > > I don't understand why cant he just post updates for the plugin but unofficially on some different github
> > 
> > 
> > He didn't remove the plug-in from people's folder because of new BetterDiscord TOS. BD's admins themselves said they didn't require him to do so.
> > One of their staff told me he was done being mentioned about this specific plug-in, so it was most likely the reason he removed it from everyone.
> > I think anyway he didn't acted in a very mature way, but that's whatever let's just move on from this guy
> 
> The plugin has broken on multiple occasions over span of past few months and his reactions to open issues didn't seem the kindest, making updates take a while too..

That's that happens when you have to maintain over 50 plug-ins and are not emotionally prepared to handle it I guess. It can quickly overwhelm you

**Comment by Minixsas** [comment]
> > > > I don't understand why cant he just post updates for the plugin but unofficially on some different github
> > > 
> > > 
> > > He didn't remove the plug-in from people's folder because of new BetterDiscord TOS. BD's admins themselves said they didn't require him to do so.
> > > One of their staff told me he was done being mentioned about this specific plug-in, so it was most likely the reason he removed it from everyone.
> > > I think anyway he didn't acted in a very mature way, but that's whatever let's just move on from this guy
> > 
> > 
> > no they did remove it if you have that plugin in version 3.1.1 and start discord it will wipe the whole file of the plugin and it will make it to write the message they literally force wiped the plugin from your pc
> 
> That's the guy who made the plug-in who did it, not BD's admins

but they forced him to do it :/

**Comment by Minixsas** [comment]
> I've noticed MessageLogger V2 has not been updated since 02.03.2021 (at least what their website says) So is it just the way it works that it does not get broken by Discord updates or I'm missing something?

Yeah most likely not cuz they have their own custom library and shit created that's why it works always

**Comment by teisseire117** [comment]
> I've noticed MessageLogger V2 has not been updated since 02.03.2021 (at least what their website says) So is it just the way it works that it does not get broken by Discord updates or I'm missing something?

dev must be sleeping well

**Comment by Minixsas** [comment]
mby we can ask them for help

**Comment by Saalvage** [comment]
Well at current time I don't think there is much to be done besides networking, the plugin works, and as long as it'll not be broken intentionally I think it'll continue working for quite some time.

Pinging @1Lighty, maybe they can share some insights on how it is developing BD plugins "under the radar". ^^

**Comment by mutageneral** [comment]
I'm currently spreading the good message in DMs in the BD discord server if you know what I mean 😎 🤫

**Comment by Oondanomala** [comment]
Because every reddit post has been deleted (including a lot of comments), here is an archive of them.
https://archive.ph/KQskg
https://archive.ph/LwRkp
https://archive.ph/VZZtN
Man i love archiving totally non controversial stuff

**Comment by mutageneral** [comment]
Someone made a Discord server for r/OpenBetterDiscord related topics: https://discord.gg/hGxNHPxQQn

**Comment by teisseire117** [comment]
> I'm currently spreading the good message in DMs in the BD discord server if you know what I mean 😎 🤫

I received it lmao 
![cxgFT07](https://user-images.githubusercontent.com/43145883/188523980-ee889852-1306-4ba2-a731-99659129576d.png)

**Comment by mutageneral** [comment]
Found another person we could potentially collaborate with!
https://github.com/JustOptimize/return-seeHiddenMessages
This guy actually forked the 0BDFDB lib and changed its name and metadata like I suggested on doing.
Ping @JustOptimize 👋

**Comment by mutageneral** [comment]
(what I love about this is how everyone so far went around fixing it in a completely different way, lol)

**Comment by Oondanomala** [comment]
I'm done with archiving reddit threads, so if anything else pops up make sure to archive it (or not, depending on what it is) ;)

**Comment by DraconicEssence** [comment]
> 

All the help is appreciated where ever. We all need to help each other during this complete and utter mess.

**Comment by BackPal** [comment]
For anyone who isn't sure how to archive, here are three extremely easy tools.

https://addons.mozilla.org/en-US/firefox/addon/archive-page/

https://chrome.google.com/webstore/detail/archive-page/gcaimhkfmliahedmeklebabdgagipbia

https://youtube-dl.org/

I believe all of these have repositories on Git as well.

**Comment by DraconicEssence** [comment]
Ah nice thanks!

**Comment by Oondanomala** [comment]
Dont use youtube-dl, [yt-dlp](https://github.com/yt-dlp/yt-dlp) is a much better fork

**Comment by Oondanomala** [comment]
I have never used a GUI for it so i have no idea which one is the best.
Im also not entirely sure what yt-dlp would be used for here but thats beside the point

**Comment by BackPal** [comment]
Just another tool for people to use in the same vein. People had been archiving pages, and archive.ph won't archive videos, so at least this provides somewhat of a way for anyone who wants to archive videos involving anything re: betterdiscord fuckery.

**Comment by rollingmoai** [comment]
There's a ShowHiddenChannels alternative already made long before without the dependency of the bdfdb library: https://github.com/BD-MemorySeal/DevilBro-Alternatives/tree/distribution/ShowHiddenChannels

**Comment by rollingmoai** [comment]
> It is pretty old, possibly completely broken as discord has updated numerous times since then. It is a good example indeed though.

Did you even try it? It works.

**Comment by Minixsas** [comment]
If that works IT would fix our main issue

**Comment by Minixsas** [comment]
Btw guys any type of fix to a certain plugin that got fucked either by its creator in alliance with BD or be itself i think its fair to say that when we fix the certain plugin original plugin creator shall not receive credit for the plugin making so edit your plugins to be names after you ( in the description) example: Author: Minixsas

**Comment by JustOptimize** [comment]
> hen we fix the certain plugin original plugin creator shall not receive credit for the plugin

You are not wrong but neither right, until the plugin is rewritten or heavily modified, I think the original creator should be credited for it

**Comment by Minixsas** [comment]
> > hen we fix the certain plugin original plugin creator shall not receive credit for the plugin
> 
> You are not wrong but neither right, until the plugin is rewritten or heavily modified, I think the original creator should be credited for it

but he shouldn't be certified for it if he forcefully fucks up his own plugin

**Comment by mutageneral** [comment]
> Btw guys any type of fix to a certain plugin that got fucked either by its creator in alliance with BD or bd itself i think its fair to say that when we fix the certain plugin original plugin creator shall not receive credit for the plugin making so edit your plugins to be names after you ( in the description) example: Author: Minixsas

You have to follow the license. As long as you respect the license, you are free to do what you want. I recommend you read the license of projects you want to fork or contribute to before you do.

**Comment by HistoricTheater** [comment]
> 
> You have to follow the license. As long as you respect the license, you are free to do what you want. I recommend you read the license of projects you want to fork or contribute to before you do.

You forfeit any right to a license when you convert your software to malware that deletes/alters files on a users system without their consent and continues to alter their system if you dont comply with their new regime.

**Comment by Saalvage** [comment]
My opinion: Even if you don't respect the person, respect the license, assuming it's permissive.

If everyone thought that way we'd end up with a huge mess of license violations with no visible gain.

**Comment by rollingmoai** [comment]
Here is another ShowHiddenChannels plugin rewritten from scratch that **doesn't require BDFDB**: https://github.com/Tharki-God/BetterDiscordPlugins/blob/master/ShowHiddenChannels.plugin.js. @phoriah Are you willing to do a cooperative request with this guy?

**Comment by Weblure** [comment]
> It wasn't uncalled for, there were reasons behind it. But after all, its free stuff. We are not entitled to it by any means

They're entitled to stop working on it, but they aren't entitled to forcibly destroy working code on other people's computers without their knowledge or consent. That's like an artist giving out free copies of their work at a fair, and then later breaking into people's homes to scribble all over it.
